src/docfx/build/toc/TableOfContentsMap.cs

```diff
@@ -75,7 +75,7 @@ public Document GetNearestToc(Document file)
             var fileNames = Path.GetFileNameWithoutExtension(file.SitePath).Split(wordSplitChars, StringSplitOptions.RemoveEmptyEntries);
             var tocCandidates = from toc in filteredTocs
                                 let dirInfo = GetRelativeDirectoryInfo(file, toc)
-                                where hasReferencedTocs || dirInfo.parentDirectoryCount >= dirInfo.subDirectoryCount
+                                where hasReferencedTocs || (dirInfo.parentDirectoryCount >= dirInfo.subDirectoryCount && dirInfo.subDirectoryCount == 0)
```

Comment:
```suggestion
                                where hasReferencedTocs || dirInfo.subDirectoryCount == 0
```
parentDirectoryCount is always positive xD